common/cache/metricsScopeCache.go

```diff
@@ -26,47 +26,96 @@ import (
 	"bytes"
 	"strconv"
 	"sync"
+	"sync/atomic"
+	"time"
 
 	"github.com/uber/cadence/common/metrics"
 )
 
+const flushBufferedMetricsScopeDuration = 10 * time.Second
+
 type domainMetricsScopeCache struct {
-	sync.RWMutex
 	scopeMap map[string]metrics.Scope
+	buffer   *buffer
+}
+
+type buffer struct {
+	sync.RWMutex
+	bufferMap map[string]metrics.Scope
 }
 
+type metricsScopeMap map[string]metrics.Scope
+
+var cache atomic.Value
```

Comment:
is this cache shared by multiple DomainMetricsScopeCache? if not, better to attach it to DomainMetricsScopeCache, instead of global

Reply:
+1 for attaching it to `DomainMetricsScopeCache`. We may have multiple scope caches for different parts of the system in the future.